src/simulators/overview_simulator.ts

```diff
@@ -0,0 +1,84 @@
+import { message } from '../shared/proto/messages'
+import { vec2$Properties } from '../shared/proto/messages'
+import { Simulator } from './simulator'
+import { Message } from './simulator'
+import Overview = message.support.nubugger.Overview
+import Mode = message.input.GameState.Data.Mode
+import Phase = message.input.GameState.Data.Phase
+import PenaltyReason = message.input.GameState.Data.PenaltyReason
+import State = message.behaviour.Behaviour.State
+
+export class OverviewSimulator implements Simulator {
+  constructor(private random: () => number) {
+  }
+
+  public static of() {
+    return new OverviewSimulator(Math.random.bind(Math))
+  }
+
+  public simulate(time: number): Message[] {
+    const messageType = 'message.support.nubugger.Overview'
+
+    var robotPosition = this.randomFieldPosition()
+    var robotHeading = this.randUnitVector()
+    var ballPosition = this.randomFieldPosition()
+    var ballWorldPosition = this.randomFieldPosition()
+
+    const buffer = Overview.encode({
+      roleName: 'Overview Simulator',
+      voltage: this.randFloat(10, 13),
+      battery: this.random(),
+      behaviourState: State.INIT,
+      robotPosition,
+      robotPositionCovariance: {
+        x: { x: this.random(), y: this.random() },
+        y: { x: this.random(), y: this.random() },
+      },
+      robotHeading,
+      ballPosition,
+      ballWorldPosition,
+      gameMode: Mode.NORMAL,
+      gamePhase: Phase.INITIAL,
+      penaltyReason: PenaltyReason.UNPENALISED,
+      lastCameraImage: { seconds: time - 5000 * this.random() },
+      lastSeenBall: { seconds: time - 5000 * this.random() },
+      lastSeenGoal: { seconds: time - 15000 * this.random() },
+      lastSeenObstacle: { seconds: time - 15000 * this.random() },
+      pathPlan: [
```

Comment:
The path plan doesn't exist in that form any more no. But we do have a ball approach position point that we could use.